### ISSUE TITLE
BUG: UnboundLocalError: local variable 'mu' referenced before assignment

### ISSUE THREAD

**Issue by jf2** [statsmodels/statsmodels#3945]
```
y = np.random.random(size=10)
model = sm.tsa.AR(y).fit(maxlag=1, method="mle", trend="nc", disp=False)
model.fittedvalues
```

Causes a: `UnboundLocalError: local variable 'mu' referenced before assignment`

The error appears in tsa/ar_model.py (line 224). Since k_trend is zero, mu is never set.
Adding an else clause to line 219 with mu=0 resolved the issue on my end.

```
if k_trend:
    mu = params[0]/(1-np.sum(params[k_trend:]))
else:
    mu = 0
```

**Comment by josef-pkt** [comment]
@jf2 Thanks for the report

This needs to be fixed for 0.9. 
There might/should be an existing test case that we can reuse for this.

**Comment by josef-pkt** [comment]
fixed in #4493

@jf2  Thanks for reporting and showing the correction
